### ISSUE TITLE
Tabs in titlebar

### ISSUE THREAD

**Issue by tim-sueberkrueb** [tim-sueberkrueb/material-browser#10]
Merge the row of tabs with the window's titlebar.

**Comment by pierremtb** [comment]
You can make a frameless window, with a custom window decoration (as I previously did on my  calculator). However, the code I used to make it dragabble was working, but it wasn't really smooth. I can try to implement it on your browser tonight if I got time.

**Comment by tim-sueberkrueb** [comment]
@pierremtb Great! Development on this goes here: https://github.com/tim-sueberkrueb/material-browser/tree/feature-10

**Comment by pierremtb** [comment]
I got some news, I created a branch for this feature : feature/custom_frame.
![capture du 2015-09-17 18 15 29](https://cloud.githubusercontent.com/assets/10795683/9939041/7743997c-5d68-11e5-8125-0b6630e9601e.png)
![capture du 2015-09-17 18 15 05](https://cloud.githubusercontent.com/assets/10795683/9939042/77476976-5d68-11e5-8fe8-8751fb1ba393.png)

**Comment by tim-sueberkrueb** [comment]
Need some work regarding subpages:
![bildschirmfoto vom 2015-10-01 12 51 46](https://cloud.githubusercontent.com/assets/11774614/10218960/4669a494-683b-11e5-97ff-b06d79687a17.png)

**Comment by pierremtb** [comment]
@tim-sueberkrueb Yes I saw that yesterday. I need to adapt the margin that I need to set for the main page

**Comment by jmc-figueira** [comment]
Amazing work! Two issues though are that I feel like the maximize and minimize buttons aren't clear from a UX perspective (usually the standard is an "underscore" for minimize and a square or a plus for maximize), although that might be the Material Design approach, and the fact that maximize does leave a small border around the window, not fully maximizing it (at least on OS X). There's also a fair bit of lag between pressing a button and the action actually being performed, as well as some glitchiness when moving a window, and notably toggling the feature in the settings only works after a restart (when going from the custom frame to the OS-default one, it will result in a frameless, control-less window).

**Comment by pierremtb** [comment]
You're completely right, I just chose that set because they were included
in QML-Material, so I had nothing to do. But they are not user friendly at
all. BTW what would you think of putting them on the left corner on OS X ?
Maybe it would be better for consistency.
The custom frame still need to be improved indeed. Regarding the need of
restarting, it is apparently OS X related, or at least non-Linux, because
it does work instantly on my eOS. Will think about it. Thank you !

**Comment by jmc-figueira** [comment]
The maximization issue and restarting have both been fixed in the meantime (between the last few commits I believe). Still, the theming issue remains, and the window is undraggable.

> BTW what would you think of putting them on the left corner on OS X ?
Maybe it would be better for consistency.

Not necessarily; it would be more user-friendly on OS X, but how would you go about Linux DEs, which have different standards (Ubuntu's Unity puts controls on the left, OS X-style, but KDE puts them on the right, for instance. GNOME Shell doesn't even have minimize and maximize by default)? I believe it's best to stick with a configuration for the material frame, in my opinion, as any user who feels confused by this can opt for the OS-specific controls instead, in the options.

**Comment by tim-sueberkrueb** [comment]
![bildschirmfoto vom 2015-10-04 11 55 42](https://cloud.githubusercontent.com/assets/11774614/10267282/ea2f4fd4-6a8e-11e5-82af-73afb6adace6.png)
Address bar elevation seems to conflict :)

**Comment by pierremtb** [comment]
@jmc-figueira Just checked too : the dragging works for me on my El Capitan

**Comment by jmc-figueira** [comment]
Dragging is fixed in the latest commit, so long as the window is not maximized.

**Comment by pierremtb** [comment]
Reworked it, no more margin on top of the tabs, the sysbuttons are now "higher" than the tabs, which go "under".
![capture du 2015-10-05 19 25 44](https://cloud.githubusercontent.com/assets/10795683/10289393/5efd1034-6b9e-11e5-9e8e-de78a92de992.png)

**Comment by tim-sueberkrueb** [comment]
* [ ] fixed
![bildschirmfoto vom 2015-10-07 12 19 34](https://cloud.githubusercontent.com/assets/11774614/10334894/ba3bb3d8-6ced-11e5-93b1-67173fbbb090.png)

**Comment by pierremtb** [comment]
@tim-sueberkrueb fixed ;)

**Comment by VirusKA** [comment]
:heart_eyes:
